integration/simulation/validate_chain.go

```diff
@@ -393,38 +393,41 @@ func checkLogsReceived(t *testing.T, s *Simulation) {
 	for owner, channel := range s.LogChannels {
 		logsReceived += checkReceivedHOCAndPOCLogs(t, owner, channel)
 	}
-	if logsReceived == 0 {
+	if logsReceived < logsThreshold {
 		t.Errorf("no logs received during simulation")
 	}
 }
 
 // Checks that the owner has only received relevant logs, and only from the HOC or POC contracts.
 func checkReceivedHOCAndPOCLogs(t *testing.T, owner string, channel chan common.IDAndLog) int {
-	logsReceived := 0
+	var logsReceived []*types.Log
 
+out:
```

Comment:
Would this work to drain the queue without the break label thingy?
```
for len(channel) > 0 {
   idAndLog <-channel
   // ...
}
```

Reply:
It doesn't seem to work, no. Not sure why, maybe one to revisit in the future.